Cython/Compiler/ExprNodes.py

```diff
@@ -4399,13 +4404,15 @@ def generate_result_code(self, code):
                 self.base.pythran_result(),
                 pythran_indexing_code(self.indices)))
             return
-        buffer_entry, self.buffer_ptr_code = self.buffer_lookup_code(code)
+        buffer_entry, self.buffer_ptr_code, buffer_temps = self.buffer_lookup_code(code)
         if self.type.is_pyobject:
             # is_temp is True, so must pull out value and incref it.
             # NOTE: object temporary results for nodes are declared
             #       as PyObject *, so we need a cast
             code.putln("%s = (PyObject *) *%s;" % (self.result(), self.buffer_ptr_code))
             code.putln("__Pyx_INCREF((PyObject*)%s);" % self.result())
+            for temp in buffer_temps:
```

Comment:
This just needs to be out of the if statement scope. I'll submit a revised version once I've found where the test should go.